### PR TITLE
WireGuard: multiple fixes

### PR DESCRIPTION
**Pull request by Vic063**
Since the update of the WireGuard code now using .NLA_EXACT_LEN attributes (see https://github.com/WireGuard/WireGuard/blob/19b2fb70620d5d664549b1e9dd5f123ab1a1baf5/src/netlink.c#L32), header types of the following classes are not correctly set when encoding NLAs:

wgpeer_a_allowedips
wgpeer_allowedip
wgdevice_a_peers
wgdevice_peer
So I forced them, taken from dumps of netlink messages using the 'wg' command.
I don't know if it's really related with .NLA_EXACT_LEN attributes but I did not understand how these values are built.

It also includes a fix for Python2 and empty IPv6 endpoint address.

### CODE REVIEW

File: pyroute2/netlink/generic/wireguard.py [hunk 1]
```diff
@@ -152,6 +158,10 @@ class wgdevice_peer(nla):
                        ('WGPEER_A_ALLOWEDIPS', 'wgpeer_a_allowedips'),
                        ('WGPEER_A_PROTOCOL_VERSION', 'uint32'))
 
+            def encode(self):
+                self['header']['type'] = 0x8000
```
Comment (ffourcot) [inline pyroute2/netlink/generic/wireguard.py]: Can you try to set `nla_flags = NLA_F_NESTED` instead to overwrite encode here?

File: pyroute2/netlink/generic/wireguard.py [hunk 2]
```diff
@@ -189,7 +199,7 @@ def encode(self):
                         self['family'] = AF_INET
                         self['addr4'] = unpack('>I',
                                                inet_aton(self['addr']))[0]
-                        self['addr6'] = bytearray(4)  # Set 4 bytes to NULL
+                        self['addr6'] = b'\x00\x00\x00\x00\x00\x00\x00\x00'
```
Comment (ffourcot) [inline pyroute2/netlink/generic/wireguard.py]: Previous code was not working?

Comment (Vic063) [reply pyroute2/netlink/generic/wireguard.py]: It worked but I think NLA auto-padding was our guardian angel.

File: pyroute2/netlink/generic/wireguard.py (hunk 1 above)
Comment (Vic063) [reply pyroute2/netlink/generic/wireguard.py]: You're right. Force pushed here d82533771a43ea5debca57adf5ab2e1a492098b1

File: pyroute2/netlink/generic/wireguard.py (hunk 2 above)
Comment (Vic063) [reply pyroute2/netlink/generic/wireguard.py]: Tested with both IPv4 and IPv6 addresses on both Python versions. It works properly.

**Pull request merged**